librz/core/linux_heap_glibc.c

```diff
@@ -1485,15 +1566,176 @@ void GH(print_malloc_info)(RzCore *core, GHT m_state, GHT malloc_state) {
+ * \param core RzCore pointer
+ * \param m_arena Offset of the arena in memory
+ * \param main_arena MallocState struct for the arena in which bin are
+ */
+static void GH(print_unsortedbin_description)(RzCore *core, GHT m_arena, MallocState *main_arena) {
+	RzConsPrintablePalette *pal = &rz_cons_singleton()->context->pal;
+	rz_cons_printf("Unsorted bin @ ");
+	PRINTF_BA("0x%" PFMT64x "\n", (ut64)m_arena);
+	int chunk_cnt = GH(print_bin_content)(core, main_arena, 0);
+	rz_cons_printf("Found %d chunks in unsorted bins \n", chunk_cnt);
+}
+
+/**
+ * \brief Prints small bins description for an arena (used for `dmhd` command)
+ * \param core RzCore pointer
+ * \param m_arena Offset of the arena in memory
+ * \param main_arena Pointer to MallocState struct for the arena in which bins are
+ */
+static void GH(print_smallbin_description)(RzCore *core, GHT m_arena, MallocState *main_arena) {
+	RzConsPrintablePalette *pal = &rz_cons_singleton()->context->pal;
+	rz_cons_printf("Small bins @ ");
+	PRINTF_BA("0x%" PFMT64x "\n", (ut64)m_arena);
+	int chunk_cnt = 0;
+	int non_empty_cnt = 0;
+	for (int bin_num = 1; bin_num < NSMALLBINS; bin_num++) {
+		int chunk_found = GH(print_bin_content)(core, main_arena, bin_num);
+		if (chunk_found > 0) {
+			non_empty_cnt += 1;
+		}
+		chunk_cnt += chunk_found;
+	}
+	rz_cons_printf("Found %d chunks in %d small bins \n", chunk_cnt, non_empty_cnt);
+}
+
+/**
+ * \brief Prints large bins description for an arena (used for `dmhd` command)
+ * \param core RzCore pointer
+ * \param m_arena Offset of the arena in memory
+ * \param main_arena Pointer to MallocState struct for the arena in which bins are
+ */
+static void GH(print_largebin_description)(RzCore *core, GHT m_arena, MallocState *main_arena) {
+	RzConsPrintablePalette *pal = &rz_cons_singleton()->context->pal;
+	rz_cons_printf("Large bins @ ");
+	PRINTF_BA("0x%" PFMT64x "\n", (ut64)m_arena);
+	int chunk_cnt = 0;
+	int non_empty_cnt = 0;
+	for (int bin_num = NSMALLBINS; bin_num < NBINS - 2; bin_num++) {
+		int chunk_found = GH(print_bin_content)(core, main_arena, bin_num);
+		if (chunk_found > 0) {
+			non_empty_cnt += 1;
+		}
+		chunk_cnt += chunk_found;
+	}
+	rz_cons_printf("Found %d chunks in %d large bins \n", chunk_cnt, non_empty_cnt);
+}
+
+/**
+ * \brief Prints description of bins for main arena for `dmhd` command
+ * \param core RzCore pointer
+ * \param m_arena Offset of main arena in memory
+ * \param main_arena Pointer to Malloc state struct for main arena
+ * \param global_max_fast The largest fast bin size (used for formatting)
+ * \param format Enum to determine which type of bins to print.
+ */
+static void GH(print_main_arena_bins)(RzCore *core, GHT m_arena, MallocState *main_arena, GHT global_max_fast, HeapBinType format) {
+	rz_return_if_fail(core && core->dbg && core->dbg->maps);
+	bool demangle = rz_config_get_i(core->config, "dbg.glibc.demangle");
+	if (format == BIN_ANY || format == BIN_TCACHE) {
+		bool main_thread_only = true;
```

Comment:
is this needed?

Reply:
The reused functions from code of `dmhf` and `dmht` have `demangle` variable as an argument.

Reply:
I mean that probably you can just pass `true` directly to the `print_tcache_instance`, without the need for a variable.